manifests/plugin/pulp.pp

```diff
@@ -17,13 +17,16 @@
 #
 # $pulpcore_content_url::  The URL to the Pulp 3 content
 #
+# $client_authentication:: An array of client authentication types supported by the Pulp installation.
+#
 class foreman_proxy::plugin::pulp (
   Foreman_proxy::ListenOn $listen_on = 'https',
   Boolean $pulpcore_enabled = true,
   Boolean $pulpcore_mirror = false,
   Stdlib::HTTPUrl $pulpcore_api_url = $foreman_proxy::plugin::pulp::params::pulpcore_api_url,
   Stdlib::HTTPUrl $pulpcore_content_url = $foreman_proxy::plugin::pulp::params::pulpcore_content_url,
   Optional[String] $version = undef,
+  Boolean $client_authentication = ['client_certificate'],
```

Comment:
```suggestion
  Array[Enum['password', 'client_certificate', 'client_certificate_admin_only']] $client_authentication = ['client_certificate'],
```

Reply:
As a general question, do we want to be coupling in these situations? Whereby an update to smart_proxy_pulp adding some new type would then force an update to this code.

Reply:
I also wondered that. The benefit of having a type here is that if you make a typo, it's caught during catalog compilation rather than smart proxy startup. On the other hand, we're not that likely to have user input so I'd also be fine with `Array[String[1], 1]` (the last 1 ensures there's at least one value, which would also be a good addition with the enum).